spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-nacos-discovery/src/main/java/com/alibaba/cloud/nacos/discovery/NacosDiscoveryClientConfiguration.java

```diff
@@ -65,15 +65,16 @@ public NacosWatch nacosWatch(NacosServiceManager nacosServiceManager,
 	}
 
 	/**
-	 * Spring Cloud Gateway HeartBeat .
+	 * Nacos HeartBeat is no longer enabled by default .
 	 * publish an event every 30 seconds
 	 * see https://github.com/alibaba/spring-cloud-alibaba/issues/2868
+	 * see https://github.com/alibaba/spring-cloud-alibaba/issues/3258
 	 */
 	@Bean
 	@ConditionalOnMissingBean
-	@ConditionalOnProperty(value = "spring.cloud.gateway.discovery.locator.enabled", matchIfMissing = false)
-	public GatewayLocatorHeartBeatPublisher gatewayLocatorHeartBeatPublisher(NacosDiscoveryProperties nacosDiscoveryProperties) {
-		return new GatewayLocatorHeartBeatPublisher(nacosDiscoveryProperties);
+	@ConditionalOnProperty(value = "spring.cloud.nacos.discovery.heart-beat.enabled", matchIfMissing = false)
```

Comment:
In this way, each user need to add relevant configuration manually, I recommend to stay current and add a configuration about Spring Boot Admin again. it won't influence some old users.